etl/steps/data/garden/smoking/2024-05-30/cigarette_sales.py

```diff
@@ -0,0 +1,168 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import numpy as np
+import owid.catalog.processing as pr
+import pandas as pd
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+COLS_WITH_DATA = [
+    "manufactured_cigarettes",
+    "manufactured_cigarettes_per_adult_per_day",
+    "handrolled_cigarettes",
+    "handrolled_cigarettes_per_adult_per_day",
+    "total_cigarettes",
+    "total_cigarettes_per_adult_per_day",
+    "all_tobacco_products_tonnes",
+    "all_tobacco_products_grams_per_adult_per_day",
+]
+
+MAN_KEY = "manufactured_cigarettes_per_adult_per_day"
+HAND_KEY = "handrolled_cigarettes_per_adult_per_day"
+TOTAL_KEY = "total_cigarettes_per_adult_per_day"
+ALL_KEY = "all_tobacco_products_grams_per_adult_per_day"
+
+
+def standardise_years(df):
+    new_df = []
+    for __, row in df.iterrows():
+        year = row["year"]
+        year_int = -1
+        dict_from_row = row.to_dict()
+        try:
+            year_int = int(year)
+            dict_from_row["year"] = year_int
+            new_df.append(dict_from_row)
+            continue
+        except ValueError:
+            if "." in year:
+                year_int = int(year.split(".")[0])
+            elif "/" in year:
+                year_int = int(year.split("/")[0])
+            if year_int > 0:
+                dict_from_row["year"] = year_int
+                new_df.append(dict_from_row)
+                continue
+            elif "-" in year:  # timeframe given in excel
+                timeframe = year.split("-")
+                start_year = int(timeframe[0])
+                end_year = int(timeframe[1])
+                if end_year < 100:
+                    if end_year > (start_year % 100):
+                        end_year = int(np.floor(start_year / 100) * 100 + end_year)
+                    elif end_year < (start_year % 100):
+                        end_year = int(np.floor(start_year / 100) * 100 + end_year + 100)
+                elif end_year > 10000:
+                    end_year = int(np.floor(end_year / 10))
+                for year_in_timeframe in range(start_year, end_year + 1):
+                    dict_from_row = row.to_dict()
+                    dict_from_row.update({"year": year_in_timeframe})
+                    new_df.append(dict_from_row)
+    return pd.DataFrame(new_df)
+
+
+def include_split_germany(tb, ds_population):
+    """Include data for Germany 1945-1990 in the table by taking weighted average of East and West Germany data"""
+    germany_tb = tb[tb["country"].isin(["West Germany", "East Germany"])]
+    germany_tb = geo.add_population_to_table(germany_tb, ds_population, interpolate_missing_population=True)
```

Comment:
I actually thought about it when doing it this way - I decided against it for two reasons:
- They are using WHO estimates for their population figures which I don't think correspond 1:1 to our population data (feel free to correct me if I am wrong here!) and I didn't know how big the effect from that change would be
- This way I am not using any absolute populations but only the share/ ratio between East/ West Germany which - even as the dataset is sparse - does not change much over that timeframe (from like 74:26 to 79:21), so I assume the effect of the interpolation to be a small error overall. 

Let me know if I have a mistake in my thinking - otherwise I would leave the function as is